script/deploy/safe/propose-to-safe.ts

```diff
@@ -106,29 +133,48 @@ const main = defineCommand({
       nonce: nextNonce,
     }
 
-    const safeTransaction = await protocolKit.createTransaction({
+    let safeTransaction = await protocolKit.createTransaction({
       transactions: [safeTransactionData],
     })
 
     const senderAddress = await signer.getAddress()
+    safeTransaction = await protocolKit.signTransaction(safeTransaction)
     const safeTxHash = await protocolKit.getTransactionHash(safeTransaction)
-    const signature = await protocolKit.signHash(safeTxHash)
 
     console.info('Signer Address', senderAddress)
     console.info('Safe Address', safeAddress)
     console.info('Network', chain.name)
     console.info('Proposing transaction to', args.to)
 
-    // Propose transaction to the service
-    await retry(async () => {
-      safeService.proposeTransaction({
+    // Store transaction in MongoDB
+    try {
+      const txDoc = {
         safeAddress: await protocolKit.getAddress(),
-        safeTransactionData: safeTransaction.data,
+        network: args.network.toLowerCase(),
+        chainId: chain.id,
+        safeTx: safeTransaction,
         safeTxHash,
-        senderAddress,
-        senderSignature: signature.data,
+        proposer: senderAddress,
+        timestamp: new Date(),
+        status: 'pending',
+      }
+
+      const result = await retry(async () => {
+        const insertResult = await pendingTransactions.insertOne(txDoc)
```

Comment:
I am not deeply in this topic so could you explain me why do we store pending transactions instead of using safe transactions api for it?

Reference: https://docs.safe.global/reference-sdk-starter-kit/safe-client/getpendingtransactions

Reply:
This is so that we have more control. There are times where the SAFE TX APIs are down or unreachable. This way we don't depend on a bunch of third-party APIs. As long as the contracts exist we can still operate.